lib/src/main/java/com/cloudbees/groovy/cps/CpsTransformer.java

```diff
@@ -239,8 +240,9 @@ CpsFunction ___cps___N() {
                 new StaticMethodCallExpression(m.getDeclaringClass(), cpsName, new TupleExpression()));
 //                new ConstructorCallExpression(FUNCTION_TYPE, new TupleExpression(params, body)));
 
-        List<Expression> paramExpressions = new ArrayList<>();
-        for (Parameter p : m.getParameters()) {
+        Parameter[] pms = m.getParameters();
+        List<Expression> paramExpressions = new ArrayList<>(pms.length);
+        for (Parameter p : pms) {
```

Comment:
It looks like all the changes to this file are some unrelated optimization. If so, would better be filed as a separate PR.